### PR TITLE
community[patch]: truncate zhipuai `temperature` and `top_p` parameters to [0.01, 0.99]

### PR DESCRIPTION
**Pull request by Congyuwang**
ZhipuAI API only accepts `temperature` parameter between `(0, 1)` open interval, and if `0` is passed, it responds with status code `400`.

However, 0 and 1 is often accepted by other APIs, for example, OpenAI allows `[0, 2]` for temperature closed range.

This PR truncates temperature parameter passed to `[0.01, 0.99]` to improve the compatibility between langchain's  ecosystem's and ZhipuAI (e.g., ragas `evaluate` often generates temperature 0, which results in a lot of 400 invalid responses). The PR also truncates `top_p` parameter since it has the same restriction.

Reference: [glm-4 doc](https://open.bigmodel.cn/dev/api#glm-4) (which unfortunately is in Chinese though).

### CODE REVIEW

**Comment by Congyuwang** [comment]
This PR follows up #16695

**Comment by Congyuwang** [comment]
@baskaryan

**Comment by Congyuwang** [comment]
Got the lint fixed.

**Review by baskaryan — APPROVED** [review]

**Comment by Congyuwang** [comment]
Looks like the workflows needs approval too. Thx.

File: libs/community/langchain_community/chat_models/zhipuai.py [hunk 1]
```diff
@@ -148,6 +148,20 @@ def _convert_delta_to_message_chunk(
     return default_class(content=content)
 
 
+def _truncate_params(payload: Dict[str, Any]) -> None:
+    """Truncate temperature and top_p parameters between [0.01, 0.99].
+
+    ZhipuAI only support temperature / top_p between (0, 1) open interval,
+    so we truncate them to [0.01, 0.99].
```
Comment (Congyuwang) [inline libs/community/langchain_community/chat_models/zhipuai.py:155]: fixed typo `truncate`

File: libs/community/langchain_community/chat_models/zhipuai.py [hunk 2]
```diff
@@ -213,7 +227,7 @@ def _default_params(self) -> Dict[str, Any]:
     model_name: Optional[str] = Field(default="glm-4", alias="model")
     """
     Model name to use, see 'https://open.bigmodel.cn/dev/api#language'.
-    or you can use any finetune model of glm series.
+    Alternatively, you can use any fine-tuned model from the GLM series.
```
Comment (Congyuwang) [inline libs/community/langchain_community/chat_models/zhipuai.py:230]: Fix some grammar in comment too.

**Pull request merged**